crates/sui-core/src/unit_tests/authority_tests.rs

```diff
@@ -293,6 +293,103 @@ async fn test_dry_run_no_gas_big_transfer() {
+            system_packages: vec![],
+        }),
+        TransactionKind::Genesis(GenesisTransaction { objects: vec![] }),
+        TransactionKind::ConsensusCommitPrologue(ConsensusCommitPrologue {
+            epoch: 0,
+            round: 0,
+            commit_timestamp_ms: 0,
+        }),
+        TransactionKind::AuthenticatorStateUpdate(AuthenticatorStateUpdate {
+            epoch: 0,
+            round: 0,
+            new_active_jwks: vec![],
+            authenticator_obj_initial_shared_version: SequenceNumber::from(0),
+        }),
+        TransactionKind::EndOfEpochTransaction(vec![]),
+        TransactionKind::RandomnessStateUpdate(RandomnessStateUpdate {
+            epoch: 0,
+            randomness_round: 0,
+            random_bytes: vec![],
+            randomness_obj_initial_shared_version: SequenceNumber::from(0),
+        }),
+        TransactionKind::ConsensusCommitPrologueV2(ConsensusCommitPrologueV2 {
+            epoch: 0,
+            round: 0,
+            commit_timestamp_ms: 0,
+            consensus_commit_digest: ConsensusCommitDigest::ZERO,
+        }),
+    ];
+
+    // Match on all the transaction kinds here so that we know we need to update this test when a
+    // new transaction type is added.
+    match &valid_transaction_kinds[0] {
+        TransactionKind::ProgrammableTransaction(_)
+        | TransactionKind::ChangeEpoch(_)
+        | TransactionKind::Genesis(_)
+        | TransactionKind::ConsensusCommitPrologue(_)
+        | TransactionKind::AuthenticatorStateUpdate(_)
+        | TransactionKind::EndOfEpochTransaction(_)
+        | TransactionKind::RandomnessStateUpdate(_)
+        | TransactionKind::ConsensusCommitPrologueV2(_) => (),
+    }
+
+    (valid_transaction_kinds, invalid_transaction_kinds)
+}
+
+#[tokio::test]
+async fn test_dev_inspect_transaction_types() {
+    let (valid_transaction_kinds, invalid_transaction_kinds) = transaction_types();
+
+    let (sender, _): (_, AccountKeyPair) = get_key_pair();
+    let gas_object_id = ObjectID::random();
+    let (_validator, fullnode, _) =
+        init_state_with_ids_and_object_basics_with_fullnode(vec![(sender, gas_object_id)]).await;
+
+    for valid_tx in valid_transaction_kinds.into_iter() {
+        let rgp = fullnode.reference_gas_price_for_testing().unwrap();
+        let res = fullnode
+            .dev_inspect_transaction_block(sender, valid_tx, Some(rgp))
+            .await
+            .unwrap();
+        assert!(res.error.is_none());
+    }
+
+    for invalid_tx in invalid_transaction_kinds.into_iter() {
+        assert!(invalid_tx.is_system_tx());
+        let rgp = fullnode.reference_gas_price_for_testing().unwrap();
+        let res = fullnode
+            .dev_inspect_transaction_block(sender, invalid_tx, Some(rgp))
+            .await;
+        assert!(res.is_err());
+        // Note this or-pattern is because the `RandomnessStateUpdate` hits an object not found
+        // error before it hits the unsupported error.
```

Comment:
that seems odd - what object is not found?

Reply:
It can't find the randomness state ID (`0x8`). 

This is because in `dev_inspect` the input objects are loaded before the transaction kind is inspected. Once the randomness object exists this will turn into the same error as the others.

Reply:
but the randomness object should exist in tests?

Reply:
Could we just match on this? so the tests don't have to be updated overtime
like `match invalid_tx { ... Random => if random_obj_exists { Unsupported } else { ObjectNotFound } ...`

Reply:
Ping on this comment and then maybe is good to go?